.github/workflows/detect_pull_request_preview.yml

```diff
@@ -1,4 +1,5 @@
 name: pr-preview-detect
+# This workflow triggers on deployment, which can never happen on a fork.
```

Comment:
This sounds like it's impossible on forks, but really it's that there's no other workflow to trigger it on forks. I presume one can trigger deployments on forks, so this workflow isn't strictly unreachable.